src/diagnostics_component/utils/file_info_utils/src/file_info_utils.c

```diff
@@ -168,13 +168,11 @@ bool FileInfoUtils_FillFileInfoWithNewestFilesInDir(FileInfo* logFiles, size_t l
 
     if (!succeeded)
     {
+        // Free all the file names, if allocated.
```

Comment:
L146: should we be doing a stat() *before* checking type?  I imagine that stat touches the file, while the S_ just are checking attribute bits.